File: client/src/app/types/course/courseDto.ts
import type { SessionDto } from "./SessionDto";

export interface CourseDto {
  id: number;
  title: string;
  description: string;
  price: number;
  subject: string;
  gradeLevel: string;
  teacherName: string;
  sessions: SessionDto[];
  quantity?: number;
}

File: API/DTOs/Course/CourseDto.cs
using API.Entities;

namespace API.DTOs.Course;

public class CourseDto
{
    public int Id { get; set; }
    public string Title { get; set; } = default!;
    public string Description { get; set; } = default!;
    public double Price { get; set; }
    public string Subject { get; set; } = default!;
    public string GradeLevel { get; set; } = default!;
    public string TeacherName { get; set; } = default!;
    public List<SessionDto> Sessions { get; set; } = [];
    //public DateTime StartDate { get; set; }
    //public DateTime EndDate { get; set; }
}
